src/c_api.rs

```diff
@@ -82,18 +82,20 @@ impl<'a> JSONApiKey<'a> {
         ctx: *mut rawmod::RedisModuleCtx,
         key_str: *mut rawmod::RedisModuleString,
     ) -> Result<JSONApiKey<'a>, RedisError> {
+        let key = RedisString::new(ctx, key_str);
         let ctx = Context::new(ctx);
-        let key = ctx.open_with_redis_string(key_str);
+        let key = ctx.open_key_writable(&key);
         JSONApiKey::new_from_key(key)
     }
 
     pub fn new_from_str(
         ctx: *mut rawmod::RedisModuleCtx,
         path: *const c_char,
     ) -> Result<JSONApiKey<'a>, RedisError> {
-        let ctx = Context::new(ctx);
         let path = unsafe { CStr::from_ptr(path).to_str().unwrap() };
```

Comment:
This line not from this PR, but still would like to raise that since it is also related to UFT-8 and lossy conversion vs. error handling mentioned in this [comment](https://github.com/RedisJSON/RedisJSON/pull/394/files#r655953307)
`let path = unsafe { CStr::from_ptr(path).to_str().unwrap() };`
This would panic if invalid UTF-8 is passed here.

Reply:
Since c_api.rs currently has no error reporting capability, we should detect conversion failure and the caller should return `null` on failure (assuming invalid UFT-8 can arrive from redis/from callers of `new_from_str`)